eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java

```diff
@@ -261,17 +263,26 @@ private void handleWriteRowsEvent(Event event, Optional<BinlogFileOffset> starti
 
       BinlogEntry entry = extractor.extract(schemaAndTable, eventData, binlogFilename, offset);
 
-
       if (!shouldSkipEntry(startingBinlogFileOffset, entry.getBinlogFileOffset())) {
         binlogEntryHandlers
-              .stream()
-              .filter(bh -> bh.isFor(schemaAndTable))
-              .forEach(binlogEntryHandler -> binlogEntryHandler.publish(entry));
+                .stream()
+                .filter(bh -> bh.isFor(schemaAndTable))
+                .forEach(binlogEntryHandler -> publish(entry, binlogEntryHandler, event));
       }
     }
 
     onEventReceived();
-    saveOffset(event);
```

Comment:
I think saveOffset() should be called when no binlogEntryHandler is invoked

Reply:
ok, will do